libyaml_q_declaration.md

```diff
@@ -0,0 +1,124 @@
+
+8.  *Must have confirmation of contributor origin (e.g. [DCO](https://developercertificate.org/), CLA, etc.)*
+    
+Does not have it (or it does not seem like it’s the case)
+
+9.  *Must have peer review policy for all change requests (e.g. require one or more reviewers)*
+    
+Seems to be followed for pull requests on the github repository, but as not all code changes occur through change requests, this can not be confirmed for these changes.
+
+10.  *Must have Continuous Integration (CI) policy for all change requests*    
+
+If any, it not publicly available
+
+11.  *Must have documentation policy for all change requests*
+    
+Not available
+
+### Documentation
+
+12.  *Must have documentation for each "feature" (e.g. for rclcpp: create a node, publish a message, spin, etc.)*
+    
+Provided [doxygen documentation](https://github.com/yaml/libyaml/tree/master/doc) for the whole project. It is not provided as a document, it has to be built separately when downloading.
+
+13.  *Must have documentation for each item in the public API (e.g. functions, classes, etc.)*
+    
+Yes, doxygen documentation is available for library .h headers.
+
+14.  *Must have a declared license or set of licenses*
+    
+MIT license declared for the repository, it can be found [here](https://github.com/yaml/libyaml/blob/master/LICENSE).
+
+15.  *Must have a copyright statement in each source file*
+    
+Is not available
+ 
+16.  *Must have a "quality declaration" document, which declares the quality level and justifies how the package meets each of the requirements*
+    
+This document represents the Quality Declaration document for the `libyaml` ROS dependency.
+
+### Testing
+
+17.  *Must have system tests which cover all items in the "feature" documentation*
+    
+Tests provided to cover the expected usage of the library, for the version of the library used can be found [here](https://github.com/yaml/libyaml/tree/release-0.1.8/tests).
+
+18.  *Must have system, integration, and/or unit tests which cover all of the public API*
+   
+Not clear without coverage results to check if all the API is covered
+
+19.  *Must have code coverage, and a policy for changes*
+
+ Code coverage and internal policies are not public, if any
+
+20.  *Performance tests, and performance regression policy*
+    
+Performance tests, and performance regression policy are not public, if any
+
+21.  *Linters and Static Analysis*
+    
+Not available publicly, if any
+
+### Dependencies
+
+22.  *Must not have direct runtime "ROS" dependencies which are not at the same level as the package in question ('Level N'), but…*
+    
+The `libyaml` library does not add additional dependencies, it only requires C++ standard libraries to be built and used.
+
+### Platform Support:
+
+23.  *Must support all tier 1 platforms for ROS 2, as defined in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers)*
+    
+Does not define support for any specific platform.
```

Comment:
Add the table https://github.com/ros2/rcutils/blob/0f448769ee02bb38231cd7656e9e75dc5939212d/QUALITY_DECLARATION.md

Reply:
I think the table is specially useful for ros2 packages, as 1. We will aim to upgrade them and 2. It summarizes why the package is qualified for a specific level.

For external dependencies though we talked in the past that the Quality Declaration file was included with the current state, but the main decision is taken on a "case by case" basis, and it's more important the extra summary section at the beginning of the file. Based on this, I think external dependencies shouldn't include this table, but I'm open to ideas on this.

@chapulina can you give us your thoughts on this case?